pkg/scheduler/partition.go

```diff
@@ -1051,7 +1022,9 @@ func (pc *PartitionContext) addAllocation(alloc *objects.Allocation) error {
 	node.AddAllocation(alloc)
 	app.RecoverAllocationAsk(alloc.Ask)
 	app.AddAllocation(alloc)
-	pc.allocations[alloc.UUID] = alloc
+
+	// track the number of allocations
+	pc.allocations += 1
```

Comment:
Any reasons for not using updateAllocationCount()? Similarly, in other places too.